### PR TITLE
#243: Convert `sharding` (non-kafka) to pekko

### PR DESCRIPTION
**Pull request by sam-byng**
incubator-pekko issue : https://github.com/apache/incubator-pekko/issues/243

**Method:** See migration guide MR https://github.com/apache/incubator-pekko/pull/249

### CODE REVIEW

File: pekko-sample-sharding-scala/build.sbt
Comment (mdedetrich) [inline pekko-sample-sharding-scala/build.sbt:1]: Can we change the `pekkoVersion` to `0.0.0+26623-85c2a469-SNAPSHOT`?

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Few minor changes but otherwise looking good. Also note that the code is not formatted, see

https://github.com/apache/incubator-pekko-samples/actions/runs/4533098550/jobs/8118457834?pr=25 specifically

```
pekko-sample-sharding-java/build.sbt
pekko-sample-sharding-scala/build.sbt
```

**Comment by sam-byng** [comment]
> Few minor changes but otherwise looking good. Also note that the code is not formatted, see
> 
> https://github.com/apache/incubator-pekko-samples/actions/runs/4533098550/jobs/8118457834?pr=25 specifically
> 
> ```
> pekko-sample-sharding-java/build.sbt
> pekko-sample-sharding-scala/build.sbt
> ```

have re-formatted now

**Review by mdedetrich — APPROVED** [review]
lgtm. Some more additional things need to be done but it can be done in future PR's.

@sam-byng BTW I just noticed that your gpg/pgp signature for git commits is coming up as not verified in github.

**Comment by sam-byng** [comment]
Thanks - note I can't merge as I don't have write access.

**Pull request merged**
